src/main/java/duke/duke/Storage.java

```diff
@@ -0,0 +1,68 @@
+package duke.duke;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.util.ArrayList;
+
+
+public class Storage {
+
+    protected String filePath;
+
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    /**
+     * Returns an ArrayList after reading from 'data/duke.ser'.
+     * Then, creates an array with all previous tasks.
+     *
+     * @return Old task list.
+     * @throws IOException If file not found.
+     * @throws ClassNotFoundException If unable to find Task class to cast ArrayList to.
+     */
+
+    public ArrayList<Task> load() throws IOException, ClassNotFoundException {
+        FileInputStream readData = new FileInputStream("data/duke.ser");
+        ObjectInputStream readStream = new ObjectInputStream(readData);
+        @SuppressWarnings("unchecked")
+        // The object read will always be an ArrayList as it always loads from the previously saved ArrayList.
+        ArrayList<Task> arr = (ArrayList<Task>) readStream.readObject();
```

Comment:
Although arr is a temporarily declared variable scoped within this method only, you may want to rename it to be more descriptive for better readability? May seem unnecessarily tedious but may prove to be helpful in the future!